catalog/files/api/src/main/java/org/projectnessie/catalog/files/api/ObjectIOExceptionMapper.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.files.api;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import org.projectnessie.nessie.immutables.NessieImmutable;
+
+@NessieImmutable
+public abstract class ObjectIOExceptionMapper {
+
+  protected abstract List<Analyzer> analyzers();
+
+  protected abstract Duration retryAfterThrottled();
+
+  protected abstract Duration retryAfterNetworkError();
+
+  protected abstract Duration reattemptAfterFetchError();
+
+  protected abstract Clock clock();
+
+  public static ImmutableObjectIOExceptionMapper.Builder builder() {
```

Comment:
Let's use a separate builder interface for this

Reply:
done